### PR TITLE
Prettify table/database errors

### PR DESCRIPTION
**Pull request by mberhault**
Introduce descriptorKey so that we can keep the db/table name around
and use it in errors.

Add TypeName to db/table descriptor for clearer error messages.

### CODE REVIEW

File: sql/database.go [hunk 1]
```diff
@@ -19,11 +19,25 @@ package sql
...
 import (
 	"github.com/cockroachdb/cockroach/keys"
+	"github.com/cockroachdb/cockroach/proto"
 	"github.com/cockroachdb/cockroach/security"
 	"github.com/cockroachdb/cockroach/sql/parser"
 	"github.com/cockroachdb/cockroach/structured"
 )
 
+// databaseKey implements descriptorKey.
+type databaseKey struct {
```
Comment (tamird) [inline sql/database.go]: `type databaseKey string`?

File: sql/descriptor.go [hunk 2]
```diff
@@ -46,7 +55,8 @@ type descriptorProto interface {
 
 // writeDescriptor takes a Table or Database descriptor and writes it
 // if needed, incrementing the descriptor counter.
-func (p *planner) writeDescriptor(key proto.Key, descriptor descriptorProto, ifNotExists bool) error {
+func (p *planner) writeDescriptor(plainKey descriptorKey, descriptor descriptorProto, ifNotExists bool) error {
+	key := plainKey.Key()
```
Comment (tamird) [inline sql/descriptor.go]: inline below?

Comment (MBerhault) [inline sql/descriptor.go]: we use it twice, hence the `key` variable.

File: sql/database.go (hunk 1 above)
Comment (MBerhault) [inline sql/database.go]: I'd rather now, there's no obvious value for it (`typename name` or `name`)

**Comment by tamird** [comment]
LGTM

**Comment by petermattis** [comment]
Nice. Those other error messages were irritating my eyes.

**Pull request merged**